checker/src/main/java/org/checkerframework/checker/nullness/CollectionToArrayHeuristics.java

```diff
@@ -140,30 +147,58 @@ private void setComponentNullness(boolean isNonNull, AnnotatedTypeMirror type) {
      * @param receiver the expression for the receiver collection
      * @return true if the argument is handled and assume to return nonnull elements
      */
-    private boolean isHandledArrayCreation(Tree argument, String receiver) {
-        if (!(argument instanceof NewArrayTree)) {
+    private boolean isHandledArrayCreation(ExpressionTree argument, String receiver) {
```

Comment:
Also update the javadoc. We can also rename the private method if you can think of a better name. It's a bit misleading now, as not just array creations are handled.